filters/no_porn_tokenizer.py

```diff
@@ -0,0 +1,84 @@
+from sacremoses import MosesTokenizer
+from toolwrapper import ToolWrapper
+from subprocess import run, PIPE
+import logging
+import sys
+import os
+import re
+
+
+# variables used by the no_escaping function
+replacements = {"&amp;":  "&",
+                "&#124;": "|",
+                "&lt;":   "<",
+                "&gt;":   ">",
+                "&apos":  "'",
+                "&quot;": '"',
+                "&#91;":  "[",
+                "&#93;":  "]"}
+
+substrs = sorted(replacements, key=len, reverse=True)
+nrregexp = re.compile('|'.join(map(re.escape, substrs)))
+
+# Back-replacements of strings mischanged by the Moses tokenizer
+def no_escaping(text):
+    global nrregexp, replacements
+    return nrregexp.sub(lambda match: replacements[match.group(0)], text)
+
+class Tokenizer:
+    def __init__(self, command=None,  l="en"):
+        if command:
+            self.cmd = command.split(' ')
```

Comment:
`shlex.split(command)` might be better here. Then people can also pass commands like `./splitter.py --param="arg with spaces"` and they would be split correctly in two instead of four parts.